addons/sale_timesheet/models/hr_timesheet.py

```diff
@@ -166,6 +167,20 @@ def _get_timesheets_to_merge(self):
         res = super()._get_timesheets_to_merge()
         return res.filtered(lambda l: not l.timesheet_invoice_id or l.timesheet_invoice_id.state != 'posted')
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if not vals.get('so_line'):
+                continue
+            so_line = self.env['sale.order.line'].browse(vals['so_line'])
+            if so_line.product_id.service_policy == 'ordered_prepaid':
+                moves = so_line.invoice_lines.move_id.filtered(
+                    lambda m: m.move_type == 'out_invoice' and m.state == 'posted'
+                ).sorted('create_date', reverse=True)
+                if moves:
+                    vals['timesheet_invoice_id'] = moves[0].id
```

Comment:
```suggestion
                )
                if moves:
                    vals['timesheet_invoice_id'] = min(moves.ids)
```
:smile:

Reply:
I'm not sure that a larger (or lower) 'id' guarantees that the invoice was created after (or before) another. I think it's true for the sequence numbers in DB but not for the 'id' field.
e.g. We have a record(id=5) that is removed. Later on, we could create another record that will automatically get assigned to id=5 to "fill in" the gaps.